source/CMakeLists.txt

```diff
@@ -19,19 +29,24 @@ set(SRCS
 
 find_package (Threads REQUIRED) #threading
 
-#-------------------------------------------------------------------------------
-#exampleApp executable setup
-if(WIN32) #windows icon resource file, provided from /packaging directory
-    set(WINDOWS_ICON_RESOURCE "${PROJECT_BINARY_DIR}/exampleApp.icon.rc")
-endif()
-
-add_executable(exampleApp ${SRCS} ${WINDOWS_ICON_RESOURCE})
+add_executable(exampleApp MACOSX_BUNDLE ${SRCS} ${ICON_RESOURCE})
 target_link_libraries (exampleApp ${CMAKE_THREAD_LIBS_INIT}) #threading
+
 if(NOT MSVC)
     target_compile_options(exampleApp PRIVATE -Wall -Wextra -Wshadow -Wnon-virtual-dtor -Wunused -pedantic) # extra warnings
 endif()
+
+if(APPLE)
+    set_target_properties(exampleApp PROPERTIES MACOSX_BUNDLE_BUNDLE_NAME "exampleApp")
+    set_target_properties(exampleApp PROPERTIES MACOSX_BUNDLE_BUNDLE_GUI_IDENTIFIER "com.exampleApp.exampleApp")
+    set_target_properties(exampleApp PROPERTIES MACOSX_BUNDLE_ICON_FILE exampleApp.icns)
+    set_target_properties(exampleApp PROPERTIES MACOSX_BUNDLE_INFO_PLIST ${PROJECT_SOURCE_DIR}/packaging/MacOSXBundleInfo.plist.in)
+    set_source_files_properties(${PROJECT_SOURCE_DIR}/packaging/exampleApp.icns PROPERTIES MACOSX_PACKAGE_LOCATION "Resources")
+endif()
+
 set_property(TARGET exampleApp PROPERTY CXX_STANDARD 14) # we want C++14
-install(TARGETS exampleApp DESTINATION bin) #copy executable to bin folder on install
+# Bundle destination specific for Mac DMG.
+install(TARGETS exampleApp BUNDLE DESTINATION "." RUNTIME DESTINATION bin)
```

Comment:
Ok, so I tested it and you are right. 

```
Installing: /Users/stephane-lb/Projects/cleanCppProject/_build/dist/./exampleApp.app
```

I have to find a way to do it as the App will go to the bin directory in all the case except when it is packaging the app in a DMG. So you have any idea on how to to do this ?